src/LoginCommand.php

```diff
@@ -434,15 +434,17 @@ private function makeMagicUrl(WP_User $user, $expires, $redirect_url)
      */
     private function persistMagicUrl(MagicUrl $magic, $endpoint, $expires)
     {
+        // We need to hash the salt to produce a key that won't exceed the maximum of 64 bytes.
+        $key = sodium_crypto_generichash(wp_salt('auth'));
```

Comment:
This should be extracted IMO and passed as a constructor arg to both places that need it.